heliomar-pena/email-server-rust
# Added CORS middleware to the project

## In this PR

Now exist the CORS middleware and is added to a new Pipeline. This Middleware checks for a "WHITELIST" environment variable which contains multiple URLs separated by commas ",".

## Screenshots

Youtube (unallowed) request:
![image](https://user-images.githubusercontent.com/66505715/196546583-62b62aef-81f0-44e1-8760-15137022025c.png)

Google (allowed) request:
![image](https://user-images.githubusercontent.com/66505715/196546623-5524ceec-5b01-45e0-b7c8-9d555a524f3f.png)

## How to test

1. Clone the repo
2. Copy the .env.example in `.env`
3. Do `cargo run`
4. Open your browser, go to any not allowed website, like www.fast.com
5. Open the console
6. Make a request to the email-microservice, something like: `fetch('http://127.0.0.1:7878').then(res => console.log(res));` 
7. Check the response code and body

## TO DO

- [x] Add status code to the response
